### PR TITLE
Topic/windows

### PR DESCRIPTION
**Pull request by HuwCampbell**
Some of the code is a little ugly still, but this seems to be functioning well.

### CODE REVIEW

**Comment by tranma** [comment]
TLDR for the silly...? what's the difference between this and the "windowed" context?

**Comment by HuwCampbell** [comment]
It's on the Windowed, but changes months to actually reflect months, and simplifies the SWindow type.

**Comment by tranma** [comment]
ah, before if was just handling days?

**Comment by HuwCampbell** [comment]
Yeah, months was just 30 days.

File: src/Icicle/Avalanche/FromCore.hs [hunk 1]
```diff
@@ -183,20 +184,36 @@ insertStream namer inputType strs reds (n, strm)
...
-               check  | Just o' <- olderThan
+               (>~~) = prim2 (PrimMinimal $ Min.PrimRelation Min.PrimRelationGt DateTimeT)
+               infix 4 >~~
+
+               (>=~~) = prim2 (PrimMinimal $ Min.PrimRelation Min.PrimRelationGe DateTimeT)
+               infix 4 >=~~
+
+               (<=~~) = prim2 (PrimMinimal $ Min.PrimRelation Min.PrimRelationLe DateTimeT)
+               infix 4 <=~~
+
```
Comment (amosr) [inline src/Icicle/Avalanche/FromCore.hs]: Maybe we can split all this out to a separate function, ``windowPredicate :: Namer n -> WindowUnit -> Maybe WindowUnit -> Exp () n Prim``

**Comment by amosr** [comment]
the Avalanche.FromCore can be cleaned up, but otherwise :+1:

File: src/Icicle/Avalanche/FromCore.hs (hunk 1 above)
Comment (tranma) [inline src/Icicle/Avalanche/FromCore.hs]: trying to compete with `lens`!

Comment (HuwCampbell) [inline src/Icicle/Avalanche/FromCore.hs]: Yeah, it's the ugliest bit. Got it working, need to make it not terrible before I merge.

**Pull request merged**